src/NHibernate/Linq/Visitors/SelectClauseNominator.cs

```diff
@@ -168,8 +171,10 @@ private bool CanBeEvaluatedInHqlSelectStatement(Expression expression, bool proj
 				return projectConstantsInHql;
 			}
 
-			// Assume all is good
-			return true;
+			return !(expression is MemberExpression memberExpression) || // Assume all is good
+			       // Evaluate only expressions that represent a mapped property
```

Comment:
Shouldn't it be "Nominate" rather than "Evaluate"?
```suggestion
			       // Nominate only expressions that represent a mapped property or a translatable method call
```

Reply:
Yes, it should.